components/Chat/ThemeSetting.tsx

```diff
@@ -0,0 +1,87 @@
+import React, { useState, useEffect } from 'react';
+import { View, TouchableOpacity, ViewStyle, StyleSheet, Switch, TextStyle } from 'react-native';
+import { AppTheme, AppConstants } from '../../config/DefaultConfig';
+import useTheme from '../../hooks/useTheme';
+import ThemedText from '../UI/ThemedText';
+import MaterialIcon from 'react-native-vector-icons/MaterialCommunityIcons';
+import useConstants from '../../hooks/useConstants';
+import { ThemeKey } from '../../config/themes';
+
+interface Props {
+  updateTheme: (theme: ThemeKey) => void;
+};
+
+const ThemeSetting: React.FunctionComponent<Props> = ({
+  updateTheme,
+}: Props) => {
+  const theme: AppTheme = useTheme();
+  const { selectedTheme }: AppConstants = useConstants();
+  const [isDarkTheme, toggleDarkTheme] = useState<boolean>(selectedTheme == ThemeKey.dark);
+
+  useEffect(() => {
+    const newSelectedTheme = isDarkTheme ? ThemeKey.dark : ThemeKey.light
+
+    updateTheme(newSelectedTheme)
+  }, [isDarkTheme]);
+
+  return (
+    <View>
+      <View style={[style.container, {borderColor: theme.lightBottomColor}]}>
+        <View style={style.leftContainer}>
+          <ThemedText styleKey="textColor">Dark Theme</ThemedText>
+        </View>
+        <View style={style.rightContainer}>
+          <Switch trackColor={{
+            false: theme.lightTextColor,
+            true: theme.lightTextColor
+          }} thumbColor={theme.textColor} value={isDarkTheme} onValueChange={toggleDarkTheme} />
+        </View>
+      </View>
+      <TouchableOpacity>
+        <View style={[style.container, {borderColor: theme.lightBottomColor}]}>
+          <View style={style.leftContainer}>
+            <ThemedText styleKey="textColor">Logout</ThemedText>
+          </View>
+          <View style={style.rightContainer}>
+            <MaterialIcon name="logout" size={30} color={theme.warningColor} />   
```

Comment:
Move logout outside.

Reply:
I have moved it outside the theme component